src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyAnalyzer.cs

```diff
@@ -101,22 +101,22 @@ protected override ExpressionSyntax GetGetterExpression(IMethodSymbol getMethod,
         protected override ExpressionSyntax GetSetterExpression(IMethodSymbol setMethod, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
             var setAccessor = setMethod.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken) as AccessorDeclarationSyntax;
-
-            // Setter has to be of the form:
-            //
-            //      set { field = value; } or
-            //      set { this.field = value; }
-            var firstStatement = setAccessor?.Body.Statements.SingleOrDefault();
-            if (firstStatement?.Kind() == SyntaxKind.ExpressionStatement)
+            var statements = setAccessor?.Body?.Statements;
+            if (statements?.Count == 1)
             {
-                var expressionStatement = (ExpressionStatementSyntax)firstStatement;
-                if (expressionStatement.Expression.Kind() == SyntaxKind.SimpleAssignmentExpression)
+                // this only works with a setter body with exactly one statement
```

Comment:
You should probably keep the original comment, which was more descriptive about what pattern the code looks for.